runtime/pom.xml

```diff
@@ -29,6 +29,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-junit5</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.jboss.weld</groupId>
```

Comment:
Why is this using Weld? Quarkus uses ArC as its CDI implementation?

If I am reading this right you want to be able to basically do injection on the client side outside the packaged application?

Is this really necessary? A normal @QuarkusIntegrationTest does not support injection on the test side, and I think adding a new injection provider into the mix could really confuse things.

Reply:
@mkouba could we possibly leverage your new QuarkusComponentTest work to allow injection into the client side of integration tests? I am not sure how compelling the use case is but I know we don't want a weld dependency.

Reply:
> Why is this using Weld? Quarkus uses ArC as its CDI implementation?
> 
> If I am reading this right you want to be able to basically do injection on the client side outside the packaged application?
> 
> Is this really necessary? A normal @QuarkusIntegrationTest does not support injection on the test side, and I think adding a new injection provider into the mix could really confuse things.

@stuartwdouglas See https://groups.google.com/g/quarkus-dev/c/HGZhNw6jsLg/m/KWptjIg6CQAJ - This was the advice given by Ladislav at the time.

Reply:
@stuartwdouglas @mkouba I have a POC based on `@QuarkusComponentTest` [here (`github.com`)]( https://github.com/turing85/quarkus-cucumber/tree/feature/support-integration-tests-with-component-test). Usability is still improvable, and we'd need to wait for a release of quarkus containing component tests, but it is a first step. Care to take a look?

Reply:
I'm not so sure it's a good idea to mix the `QuarkusComponentTestExtension` with `CucumberQuarkusIntegrationTest`. I think that what Stuart meant was to reuse some logic from the `QuarkusComponentTestExtension`, i.e. init an ArC container for the classes you want to inject in the `CucumberQuarkusIntegrationTest`. Basically some logic from the `QuarkusComponentTestExtension.initArcContainer()` and `QuarkusComponentTestExtension.afterAll()` methods.